### PR TITLE
fix(curriculum): remove all instances of teaching var in RPG game

### LINKED ISSUE #53164: Remove `var` completely from JavaScript curriculum
I know that `var` is still valid JavaScript, but many people see it early in the JavaScript curriculum and come to the wrong conclusion that our curriculum is out of date. (We introduce `let` and `const` a few steps later in the first project.)

We should simply stop teaching `var` as the confusion isn't worth it. There isn't really any practical reason to use `var` in 2024 – at least not for a beginning JavaScript developer.

### LINKED ISSUE #53187: Step 23 of RPG game is missing backticks for numbers
### Describe the Issue

In the description for step 23, it is missing backticks for the number `0`.

Here is the change needed

```md
# --description--

Create another variable called `currentWeapon` and set it to `0`.

When a variable name has multiple words, the convention in JavaScript is to use what's called <dfn>camelCase</dfn>. The first word is lowercase, and the first letter of every following word is uppercase.
```

## Note to contributors 

This has been opened up for first timers only contribution

Please read through the [contributing](https://github.com/freeCodeCamp/freeCodeCamp/blob/cedc365eb89e9c217d11ca300505fc886c966bed/CONTRIBUTING.md) docs first.
We do not assign issues except to long time contributors.
If you are interested in working on this, follow the contributing rules, fix the issue and create a PR. 

Here is the file that needs fixing 
https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/main/curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a3a7e4f1060e2fc5ffb34b.md

If you need help, then please reach out on the forum or discord channels.

Happy contributing 

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-javascript-by-building-a-role-playing-game/step-23

### Your code

see explanation above

### Expected behavior

see explanation above

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jdwilkin4**
## Summary changes
This PR is responsible for removing all instances of the `var` keyword in the RPG game. 

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #53164 and closes #53187.

### CODE REVIEW

**Comment by Kashinathpat** [comment]
@jdwilkin4 in this [file](https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a3a0a3c0a4b32915d26a6e.md) 
I think you changed variable name `xp` to `let`
```diff
-assert.match(code, /xp/);
+assert.match(code, /let/);
```

**Comment by jdwilkin4** [comment]
Hey @Kashinathpat ! 

Thanks for catching that.
I will push a commit to fix that.

Also, a neat trick you can do for PR's is to use the suggestion tool.
On GitHub, you can click on a line in the PR like this

Screenshot 2024-01-12 at 8 51 33 PM

Then click on the suggestion tool
Screenshot 2024-01-12 at 8 52 12 PM

Then you are able to make a change
Screenshot 2024-01-12 at 8 52 33 PM

And that change will show up for the author of the PR and they can commit that change right from GitHub 👍 

Screenshot 2024-01-12 at 8 53 43 PM

It has been super helpful for all of the maintainers and makes reviewing PR's much easier 👍

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8d382cd075f4169223e14.md
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8d382cd075f4169223e14.md:1]: Description in this step mentions `var`. Small rewording might be needed.

**Review by gikf — COMMENTED** [review]
`var` is still present also in few tests - in steps 110 and 165-168.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62aa27c40ca6f04ab8be5fac.md
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62aa27c40ca6f04ab8be5fac.md:23]:
```suggestion
assert.match(pick.toString(), /for\s*\(\s*let\s+x\s*=\s*1\s*;\s*x\s*<\s*5\s*;\s*x\s*\+\+\s*\)/);
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62aa28032d863d4bd8058799.md
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62aa28032d863d4bd8058799.md:21]:
```suggestion
assert.match(pick.toString(), /for\s*\(\s*let\s+i\s*=\s*0\s*;/)
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62aa287434dc284cea01157c.md
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62aa287434dc284cea01157c.md:19]:
```suggestion
assert.match(pick.toString(), /for\s*\(\s*let\s+i\s*=\s*0\s*;\s*i\s*<\s*10\s*;/)
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62aa28bbd6323e4dfb3ac43e.md
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62aa28bbd6323e4dfb3ac43e.md:19]:
```suggestion
assert.match(pick.toString(), /for\s*\(\s*let\s+i\s*=\s*0\s*;\s*i\s*<\s*10\s*;\s*i\+\+\s*\)/)
```

**Review by gikf — COMMENTED** [review]
Just some nitpicking.

**Review by gikf — APPROVED** [review]
LGTM :+1:

**Comment by gikf** [comment]
Ugh, of course, the `pick.toString()` transpiles the code and uses `var`. Because of that `code` will need to be matched.
```javascript
function pick(guess) {
  const numbers = [];
  while (numbers.length < 10) {
    numbers.push(Math.floor(Math.random() * 11));
  }
  text.innerText = "You picked " + guess + ". Here are the random numbers:\n";
  for (let x = 1; x < 5; x++) {

  }
}
console.log(pick.toString());
```

```javascript
// console
function pick(guess) {
  var numbers = [];
  var _LPC25 = 1;
  var _LP25 = Date.now();
  while (numbers.length < 10) {
    if (_LPC25++ % 100 === 0 && Date.now() - _LP25 > 100) {
      (function (e) {
        console.log("Potential infinite loop detected on line " + e + ". Tests may fail if this is not changed.");
      })(265, 2);
      break;
    }
    numbers.push(Math.floor(Math.random() * 11));
  }
  text.innerText = "You picked " + guess + ". Here are the random numbers:\n";
  var _LPC26 = 1;
  var _LP26 = Date.now();
  for (var x = 1; x < 5; x++) {
    if (_LPC26++ % 100 === 0 && Date.now() - _LP26 > 100) {
      (function (e) {
        console.log("Potential infinite loop detected on line " + e + ". Tests may fail if this is not changed.");
      })(269, 2);
      break;
    }
  }
}
```

**Comment by jdwilkin4** [comment]
> Ugh, of course, the pick.toString() transpiles the code and uses var

Yeah, that's fine.

The goal of this PR is to remove all instances of var in the lessons that the campers see.

A direct quote from the issue linked to this PR

> We should simply stop teaching var as the confusion isn't worth it.

As long as we remove all instances from the teaching lessons and just jump straight into `let`, then we are fine.

**Comment by ojeytonwilliams** [comment]
We won't be transpiling `const` and `let` for long: https://github.com/freeCodeCamp/freeCodeCamp/pull/53127, but I stalled, because I wasn't sure how best to handle the fact we're _also_ not transpiling `() => {}` and so cannot just look for `function` any more.

I'll get to that Soon(ish) if you want to match `let`.

**Comment by jdwilkin4** [comment]
> I'll get to that Soon(ish) if you want to match let.

That would be great, but I also don't think there is a rush since I don't think that should be a blocking factor for this PR. 

The purpose of the issue brought up by Quincy and this resulting PR is to remove all instances of teaching `var` to campers.

So I think it is fine, if there is a separate PR later on to refactor the tests to only match `let`

**Review by huyenltnguyen — APPROVED** [review]
LGTM.

@jdwilkin4 I'll let you handle the merge as I'm not sure if this one can land or we want to wait for #53127.

**Review by naomi-lgbt — APPROVED** [review]

**Pull request merged**
